pkg/kubelet/dockershim/docker_checkpoint.go

```diff
@@ -0,0 +1,110 @@
+package dockershim
+
+import (
+	"encoding/json"
+	"github.com/golang/glog"
+	"os"
+)
+
+const (
+	// default directory to store pod sandbox checkpoint files
+	sandboxCheckpointDir = "/var/lib/dockershim/sandbox"
+	ProtocolTCP          = Protocol("tcp")
+	ProtocolUDP          = Protocol("udp")
+	PortMappingsKey      = "PortMappings"
+	schemaVersion        = "v1"
+)
+
+type Protocol string
+
+// PortMapping is the port mapping configurations of a sandbox.
+type PortMapping struct {
+	// Protocol of the port mapping.
+	Protocol *Protocol `json:"protocol,omitempty"`
+	// Port number within the container.
+	ContainerPort *int32 `json:"container_port,omitempty"`
+	// Port number on the host.
+	HostPort *int32 `json:"host_port,omitempty"`
+}
+
+// CheckpointData contains all types of data that can be stored in the checkpoint.
+type CheckpointData struct {
+	PortMappings []*PortMapping `json:"port_mappings,omitempty"`
+}
```

Comment:
If https://github.com/containernetworking/cni/issues/353 flies, maybe we don't need this at all?